sliceanddice/provider/src/main/java/com/novoda/sliceprovider/DemoSlice.kt

```diff
@@ -0,0 +1,113 @@
+                .setPrimaryAction(sliceAction(context))
+        )
+        addRow(RowBuilder(context, sliceUri)
+            .setTitle("This is a row item title")
+            .setSubtitle("...and this is the subtitle")
+            .setTitleItem(Icon.createWithResource(context, R.drawable.ic_sun))
+            .addEndItem(sliceAction(context, true))
+        )
+        addInputRange(
+            InputRangeBuilder(this)
+                .setTitle("This is an input range item")
+                .setMax(7)
+                .setValue(5)
+                .setAction(pendingIntent(context, 86586))
+        )
+        addRange(
+            RangeBuilder(this)
+                .setTitle("This is a range item")
+                .setMax(7)
+                .setValue(2)
+        )
+        addRow(RowBuilder(context, sliceUri)
+            .setTitle("This next item is a grid item:")
+        )
+        addGrid(
+            GridBuilder(this)
+                .apply {
+                    addCell(GridBuilder.CellBuilder(this)
+                        .addImage(Icon.createWithResource(context, R.drawable.ic_android), GridBuilder.ICON_IMAGE)
+                        .addText("Icon")
+                        .addTitleText("Title text")
+                        .setContentIntent(pendingIntent(context, 8456))
+                    )
+                    addCell(GridBuilder.CellBuilder(this)
+                        .addImage(Icon.createWithResource(context, R.drawable.yucca), GridBuilder.SMALL_IMAGE)
+                        .addText("Small")
+                        .addTitleText("Title text")
+                        .setContentIntent(pendingIntent(context, 8964))
+                    )
+                    addCell(GridBuilder.CellBuilder(this)
+                        .addImage(Icon.createWithResource(context, R.drawable.dianella), GridBuilder.LARGE_IMAGE)
+                        .addText("Large")
+                        .addTitleText("Title text")
+                        .setContentIntent(pendingIntent(context, 6585))
+                    )
+                    addSeeMoreCell(GridBuilder.CellBuilder(this)
+                        .addText("See more!")
+                        .addTitleText("Title text")
+                        .setContentIntent(pendingIntent(context, 3287))
+                    )
+                }
+        )
+    }
+    .addSeeMoreRow(
+        RowBuilder(context, sliceUri)
+            .setTitle("See more row item️")
+    )
+    .addAction(sliceAction(context))
+    .addAction(sliceAction2(context))
+    .build()
+
+private fun sliceAction(context: Context, isChecked: Boolean = false) = SliceAction(
+    pendingIntent(context, 1234),
+    Icon.createWithResource(context, R.drawable.ic_android),
+    "actionnnnnn",
+    isChecked
+)
+
+private fun pendingIntent(context: Context, reqCode: Int): PendingIntent {
+    return PendingIntent.getActivity(
+        context,
+        reqCode,
+        Intent(Settings.ACTION_SETTINGS),
```

Comment:
Every time I touch a control in the slice demo it navigates to Settings and it's very annoying :/

Reply:
Yeah, well, it's meant to demonstrate that things are clickable. _In the future_ I plan to demonstrate using listeners to prevent this on the host side.